kernel/posix/semaphore.c

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <pthread.h>
+#include <errno.h>
+
+int sem_destroy(sem_t *semaphore)
+{
+	if (semaphore == NULL) {
+		errno = EINVAL;
+		return -1;
+	}
+
+	k_sem_reset(semaphore);
+
+	return 0;
+}
+
+int sem_getvalue(sem_t *semaphore, int *value)
+{
+	*value = (int) k_sem_count_get(semaphore);
+
+	return 0;
+}
+
+int sem_init(sem_t *semaphore, int pshared, unsigned int value)
+{
+	if (value == 0) {
+		errno = EINVAL;
+		return -1;
+	}
+
+	/* pshared is not supported in zephyr. For invalid
+	 * values of pshare return error
+	 */
+	if (pshared != 0) {
+		errno = ENOSYS;
+		return -1;
+	}
+
+	k_sem_init(semaphore, value, CONFIG_SEM_VALUE_MAX);
+
+	return 0;
+}
+
+int sem_post(sem_t *semaphore)
+{
+	if (semaphore == NULL) {
+		errno = EINVAL;
+		return -1;
+	}
+
+	k_sem_give(semaphore);
+	return 0;
+}
+
+static int take_and_convert(sem_t *semaphore, s32_t timeout)
+{
+	int temp;
+
+	temp = k_sem_take(semaphore, timeout);
+
+	if (temp == -EBUSY) {
```

Comment:
there is no ETIMEDOUT return in case of POSIX 1003.1 standard in case of "The sem_trywait() and sem_wait() "  - http://pubs.opengroup.org/onlinepubs/000095399/functions/sem_trywait.html

Reply:
only in case of "sem_timedwait" it can return ETIMEDOUT.
So please handle accordingly. I think no need to put it in separate function take_and_convert .